src/P3Scheme.jl

```diff
@@ -451,10 +409,500 @@ function distribution_parameter_solver(
+            large = false
+
+            v += integrate(FT(0), D_th, ai[i] * N_0, bi[i] + μ, ci[i] + λ)
+
+            # D_th to D_gr
+            if !large && th.D_gr > cutoff
+                v += integrate(
+                    D_th,
+                    cutoff,
+                    ai[i] *
+                    N_0 *
+                    (16 * p3.ρ_i^2 * p3.γ^3 / (9 * FT(π) * α_va^2))^κ,
+                    bi[i] + μ + κ * (3 * p3.σ - 2 * p3.β_va),
+                    ci[i] + λ,
+                )
+
+                # Switch to large particles
+                (ai, bi, ci) = CO.Chen2022_vel_coeffs_large(Chen2022, ρ_a)
+                large = true
+
+                v += integrate(
+                    cutoff,
+                    th.D_gr,
+                    ai[i] *
+                    N_0 *
+                    (16 * p3.ρ_i^2 * p3.γ^3 / (9 * FT(π) * α_va^2))^κ,
+                    bi[i] + μ + κ * (3 * p3.σ - 2 * p3.β_va),
+                    ci[i] + λ,
+                )
+            else
+                v += integrate(
+                    D_th,
+                    th.D_gr,
+                    ai[i] *
+                    N_0 *
+                    (16 * p3.ρ_i^2 * p3.γ^3 / (9 * FT(π) * α_va^2))^κ,
+                    bi[i] + μ + κ * (3 * p3.σ - 2 * p3.β_va),
+                    ci[i] + λ,
+                )
+            end
+
+            # D_gr to D_cr
+            if !large && th.D_cr > cutoff
+                v += integrate(
+                    th.D_gr,
+                    cutoff,
+                    ai[i] * N_0 * (p3.ρ_i / th.ρ_g)^(2 * κ),
+                    bi[i] + μ,
+                    ci[i] + λ,
+                )
+
+                # Switch to large particles
+                (ai, bi, ci) = CO.Chen2022_vel_coeffs_large(Chen2022, ρ_a)
+                large = true
+
+                v += integrate(
+                    cutoff,
+                    th.D_cr,
+                    ai[i] * N_0 * (p3.ρ_i / th.ρ_g)^(2 * κ),
+                    bi[i] + μ,
+                    ci[i] + λ,
+                )
+            else
+                v += integrate(
+                    th.D_gr,
+                    th.D_cr,
+                    ai[i] * N_0 * (p3.ρ_i / th.ρ_g)^(2 * κ),
+                    bi[i] + μ,
+                    ci[i] + λ,
+                )
+            end
+
+            # D_cr to Infinity
+            if !large
+                (I, e) = QGK.quadgk(
+                    D ->
+                        (
+                            16 *
+                            p3.ρ_i^2 *
+                            (F_r * π / 4 * D^2 + (1 - F_r) * p3.γ * D^p3.σ)^3 /
+                            (9 * π * (α_va / (1 - F_r) * D^p3.β_va)^2)
+                        )^κ *
+                        ai[i] *
+                        D^(bi[i]) *
+                        exp(-ci[i] * D) *
+                        N_0 *
+                        D^μ *
+                        exp(-λ * D),
```

Comment:
Same as my comment above, this function could be saved since it is used 3 times.